### ISSUE TITLE
ListView not working with Pjax for Gii Generator

### ISSUE THREAD

**Issue by Ariestattoo** [yiisoft/yii2-gii#200]
### What steps will reproduce the problem?
Generate a CRUD and choose Listview and Pjax

### What's expected?
When making changes to the Activeform and submitting the form a XHR request should happen 

### What do you get instead?
A full page request

### Additional info
Possible Solution
/vendor/yiisoft/yii2-gii/generators/crud/default/views/index.php 
line 36: Replacement 
<?= $generator->enablePjax ? "<?php Pjax::begin(['formSelector' => '#search-form']); ?>" : '' ?>

/vendor/yiisoft/yii2-gii/generators/crud/default/views/_search.php
line 25 : Insertion
<?php if( $generator->enablePjax){?>'id' => 'search-form',<?php }?>

However a standardised method for naming the form to Pjax connection would need to be created. Additionally an update to the documentation instructing that a form selector is required when using the listview Widget with Pjax would be helpful.

| Q                | A
| ---------------- | ---
| Yii version      | 2.09
| PHP version      | 7.0.11-1+deb.sury.org~xenial+1
| Operating system | Linux 4.4.0-38-generic #57-Ubuntu SMP x86_64

**Comment by SilverFire** [comment]
> <?php if( $generator->enablePjax){?>'id' => 'search-form',<?php }?>

It's better to add `data-pjax="1"` attribute to the form when PJAX is enabled.

Would you like to create a PR and participate in [Hacktoberfest](https://github.com/blog/2260-hacktoberfest-is-back) festival to earn a T-Shirt?

**Comment by Ariestattoo** [comment]
Hi SilverFire,
Thanks for the invite and the feedback. After spending a little bit more time with the code I have come to realise that the real issue was related to my own customised code. Using the data-pjax="1" works perfectly with a standard form submission, but I modified my code to deliver a seamless experience. I have several form inputs that have event handlers  for changes which for example when a radio button is toggled a requests is automatically sent for a new data provider. The difficulty I found was when I placed my form within the Pjax:begin was that my event handlers were broken after the first pjax request. This required a specific form selectors. I will however make a PR for the Gii generator to fix the basic functionality and appreciate the inclusion. 

Regards
